include/picongpu/particles/synchrotron/AlgorithmSynchrotron.hpp

```diff
@@ -0,0 +1,431 @@
+/* Copyright 2014-2024 Marco Garten, Jakob Trojok, Filip Optolowicz
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+#include "picongpu/algorithms/KinEnergy.hpp" // used in struct SynchrotronIdea
+#include "picongpu/param/synchrotron.param"
+
+/** @file AlgorithmSynchrotron.hpp
+ *
+ * Synchrotron ALGORITHM for the Synchrotron model
+ * Algorithm from the paper: "Extended particle-in-cell schemes for physics in ultrastrong laser fields: Review and
+ * developments" by A. Gonoskov et.Al.
+ *
+ * This file was created based on AlgorithmIonization.hpp
+ *
+ * This synchrotron extension consists of three new files:
+ *      - synchrotron.param:            Contains the parameters for the synchrotron model and initialization
+ *      - AlgorithmSynchrotron.hpp:     Defines the algorithm and data structures for the synchrotron model
+ *      - SynchrotronRadiation.hpp:     Initializes the class with precomputed F1 and F2 functions
+ * and modifies two existing files:
+ *      - Simulation.hpp:               Registers the SynchrotronRadiation stage
+ *      - ParticleFunctors.hpp:         Chooses which particles are affected by SynchrotronRadiation
+ *
+ * The call structure goes like this:
+ *     Simulation.hpp -> SynchrotronRadiation.hpp -> ParticleFunctors.hpp -> AlgorithmSynchrotron.hpp
+ *                (--)                       (++)                     (--)                       (++)
+ * (++) = containt references to the synchrotron.param file
+ *
+ * file locations:
+ * picongpu/include/picongpu/param/synchrotron.param
+ * picongpu/include/picongpu/particles/synchrotron/AlgorithmSynchrotron.hpp
+ * picongpu/include/picongpu/simulation/stage/SynchrotronRadiation.hpp
+ * picongpu/include/picongpu/simulation/control/Simulation.hpp
+ * picongpu/include/picongpu/particles/ParticleFunctors.hpp
+ */
+namespace picongpu
+{
+    namespace particles
+    {
+        namespace synchrotron
+        {
+            /** \struct SynchrotronIdea
```

Comment:
you can skip `\struct` because this comment belongs to a struct. If you like to keep it please use `@`

Reply:
done